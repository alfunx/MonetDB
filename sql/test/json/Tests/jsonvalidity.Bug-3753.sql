SELECT json.isvalid('[0,1]');
SELECT json.isvalid('[0,,1]');

SELECT json.isValid(null); 	-- false
SELECT json.isValid(''); 	-- false
SELECT json.isValid('null'); 	-- TRUE, must be false 
SELECT json.isValid('"'); 	-- false
SELECT json.isValid('""'); 	-- false
SELECT json.isValid('"""'); 	-- TRUE, must be false
SELECT json.isValid('"\""'); 	-- TRUE, must be false
SELECT json.isValid('""""'); 	-- false
SELECT json.isValid('"\"\""'); 	-- false

SELECT json.isValid('[]'); 	-- true
SELECT json.isValid('[null]'); 	-- true
SELECT json.isValid('[""]'); 	-- FALSE, must be true
SELECT json.isValid('["""]'); 	-- TRUE, must be false
SELECT json.isValid('["\""]'); 	-- true
SELECT json.isValid('[""""]'); 	-- false
SELECT json.isValid('["\"\""]'); 	-- FALSE, must be true

SELECT json.isValid('{}'); 	-- true
SELECT json.isValid('{"test":null}'); 	-- true
SELECT json.isValid('{"test":""}'); 	-- FALSE, must be true
SELECT json.isValid('{"test":"""}'); 	-- TRUE, must be false
SELECT json.isValid('{"test":"\""}'); 	-- true
SELECT json.isValid('{"test":""""}'); 	-- false
SELECT json.isValid('{"test":"\"\""}'); 	-- FALSE, must be true
