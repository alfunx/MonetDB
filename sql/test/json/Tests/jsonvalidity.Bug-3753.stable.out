stdout of test 'jsonvalidity.Bug-3753` in directory 'sql/test/json` itself:


# 23:37:49 >  
# 23:37:49 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=30311" "--set" "mapi_usock=/var/tmp/mtest-17553/.s.monetdb.30311" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/export/scratch1/mk/Jul2015//Linux/var/MonetDB/mTests_sql_test_json" "--set" "mal_listing=0" "--set" "embedded_r=yes"
# 23:37:49 >  

# MonetDB 5 server v11.22.0
# This is an unreleased version
# Serving database 'mTests_sql_test_json', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.590 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.ins.cwi.nl:30311/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-17553/.s.monetdb.30311
# MonetDB/GIS module loaded
# Start processing logs sql/sql_logs version 52200
# Start reading the write-ahead log 'sql_logs/sql/log.16'
# Finished reading the write-ahead log 'sql_logs/sql/log.16'
# Finished processing logs sql/sql_logs
# MonetDB/SQL module loaded
# MonetDB/R   module loaded

Ready.

# 23:37:49 >  
# 23:37:49 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-17553" "--port=30311"
# 23:37:49 >  

#SELECT json.isvalid('[0,1]');
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ true	]
#SELECT json.isvalid('[0,,1]');
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid(null); 	-- false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid(''); 	-- false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('null'); 	-- TRUE, must be false 
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('"'); 	-- false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('""'); 	-- false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('"""'); 	-- TRUE, must be false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('"\""'); 	-- TRUE, must be false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('""""'); 	-- false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('"\"\""'); 	-- false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('[]'); 	-- true
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ true	]
#SELECT json.isValid('[null]'); 	-- true
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ true	]
#SELECT json.isValid('[""]'); 	-- FALSE, must be true
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ true	]
#SELECT json.isValid('["""]'); 	-- TRUE, must be false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('["\""]'); 	-- true
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ true	]
#SELECT json.isValid('[""""]'); 	-- false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('["\"\""]'); 	-- FALSE, must be true
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ true	]
#SELECT json.isValid('{}'); 	-- true
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ true	]
#SELECT json.isValid('{"test":null}'); 	-- true
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ true	]
#SELECT json.isValid('{"test":""}'); 	-- FALSE, must be true
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ true	]
#SELECT json.isValid('{"test":"""}'); 	-- TRUE, must be false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('{"test":"\""}'); 	-- true
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ true	]
#SELECT json.isValid('{"test":""""}'); 	-- false
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ false	]
#SELECT json.isValid('{"test":"\"\""}'); 	-- FALSE, must be true
% .L # table_name
% isvalid_single_value # name
% boolean # type
% 5 # length
[ true	]

# 23:37:50 >  
# 23:37:50 >  "Done."
# 23:37:50 >  

